src/marqo/tensor_search/health.py

```diff
@@ -1,10 +1,108 @@
-from typing import Optional, Tuple, Union
+import math
 import copy
+from typing import Optional, Tuple, Union
 
-from marqo.config import Config
 from marqo import errors
 from marqo._httprequests import HttpRequests
+from marqo.config import Config
+from marqo.tensor_search.tensor_search_logging import get_logger
 from marqo.tensor_search.enums import HealthStatuses
+from marqo.tensor_search import constants
+from marqo.tensor_search import validation
+
+logger = get_logger(__name__)
+
+
+def convert_watermark_to_bytes(watermark: str, total_in_bytes: int = None):
+    """
+    Converts a value to bytes.
+    It could possible be:
+    1. Bytes (eg 123.4gb) - do nothing
+    2. MB, GB, TB, etc, (eg 123.4gb) - multiply by some power of 1024 to get bytes
+    3. Ratio (e.g. 0.9) - multiply by total_in_bytes to get bytes
+    4. Percentage (e.g. 90%) - convert to ratio then multiply by total_in_bytes to get bytes
+
+    Returns: watermark in bytes (int)
+    """
+
+    # Initial validation
+    if watermark is None:
+        raise errors.InternalError("OpenSearch disk watermark cannot be None.")
+    watermark = watermark.replace(" ", "")
+    if watermark == "":
+        raise errors.InternalError("OpenSearch disk watermark cannot be empty string.")
+    
+    if total_in_bytes is not None and total_in_bytes < 0:
+        raise errors.InternalError("OpenSearch cluster stats fs: total_in_bytes cannot be negative.")
+
+    if watermark[-2:].lower() in constants.BYTE_SUFFIXES:
+        # Watermark in KB/MB/GB/TB format
+        # Bytes represent MIN disk space AVAILABLE
+        # TODO: Do we use 1000 or 1024 for conversion?
+        numeric_watermark = validation.validate_nonnegative_number(watermark[:-2], "OpenSearch disk watermark value")
+        multiplier = 1024 ** constants.BYTE_SUFFIXES.index(watermark[-2:].lower())
```

Comment:
I suggest adding a comment in constants on the significance of the order (i.e. index used as the power)

Reply:
changed the byte suffixes to a dict instead